shared/lib/formatting/index.js

```diff
@@ -332,8 +335,37 @@ function unescapeAndFormatSpaces( str ) {
 	return decodeEntities( str ).replace( / /g, nbsp );
 }
 
+/**
+ * Traverse an object or array decoding HTML entities on string properties
+ * Accepts a paths argument selecting which properties to update (See lodash.get)
+ *
+ * It does its job in place, so the object passed will be updated.
+ * This helper is used to decode HTML entities in strings properties,
+ * on an object.
+ * We need to decode HTML entities because the REST API returns them
+ * already encoded, and React will encode them again.
+ *
+ * @param	{Object} obj Object to traverse in preparation for React rendering
+ * @param {String/Array} paths Path or paths to decode.
+ */
+function traverseAndDecodeEntities( obj, paths ) {
```

Comment:
My biggest concern with this method is that it mutates the original data returned from the API. While I don't think it's explicitly called out as a no-no in the docs, it is something we should avoid. See - http://calypso.localhost:3000/devdocs/docs/coding-guidelines/javascript.md#default-parameters

On another note, I think you could save yourself some work and make the code more readable by using a higher-order function such as `mapValues`. See - https://lodash.com/docs#mapValues